src/foam/nanos/column_config/ColumnConfigPropertyValue.js

```diff
@@ -0,0 +1,223 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.column',
+  name: 'ColumnConfigToPropertyConverter',
+  documentation: `ColumnConfigToPropertyConverter gathers methods for converting (or mapping) property name to property if required 
+      and returns property and obj, which will return the proprty value on f function call`,
+  javaImports: [
+    'foam.core.ClassInfo',
+    'foam.core.FObject',
+    'foam.core.PropertyInfo',
+    'foam.core.X',
+    'foam.mlang.Expr',
+    'foam.nanos.logger.Logger',
+    'java.lang.reflect.Method'
+  ],
+  methods: [
+    {
+      name: 'filterExportedProps',
+      documentation: 'returns propNames that are properties of the class and aren\'t networkTransient',
+      code: function(of, propNames) {
+        var props = of.getAxiomsByClass(foam.core.Property);
+        var allColumnNames = props.map(p => p.name);
+        if ( ! propNames ) {
+          return props.filter(p => ! p.networkTransient ).map(p => p.name);
+        } props = [];
+        return propNames.filter(n => { 
+          return allColumnNames.includes(n.split('.')[0]) && ! this.returnProperty(of, n).networkTransient;
+        });
+      }
+    },
+    {
+      name: 'returnProperty',
+      documentation: 'returns last property for chain of properties e.g. city property for address.city',
+      type: 'PropertyInfo',
+      args: [
+        {
+          name: 'of',
+          type: 'ClassInfo'
+        },
+        {
+          name: 'propName',
+          class: 'String'
+        }
+      ],
+      code: function(of, propName) {
+        var cls = of;
+        var property;
+        if ( foam.String.isInstance(propName) ) {
+          var propNames = propName.split('.');
+          for ( var i = 0 ; i < propNames.length ; i++ ) {
+            property = foam.String.isInstance(propNames[i])
+            ? cls.getAxiomByName(propNames[i])
+            :  foam.Array.isInstance(propNames[i]) ? 
+            cls.getAxiomByName(propNames[i]) : propNames[i];
+            if ( ! property )
+              break;
+            cls = property.of;
+          }
+        } else
+          property = propName;
+        return property;
+      },
+      javaCode: `
+        ClassInfo ci = of;
+        PropertyInfo p = null;
+        String[] propNames = propName.split("\\\\.");
+        StringBuilder sb = new StringBuilder("find");
+
+        for ( int i = 0 ; i < propNames.length ; i++ ) {
+          if ( ( p == null && i != 0 ) || ci == null )
+            break;
+          p = (PropertyInfo) ci.getAxiomByName(propNames[i]);
+          if ( i != propNames.length - 1 ) {
+            try {
+              Class cls;
+              if ( p instanceof foam.core.AbstractFObjectPropertyInfo ) {
+                cls = p.getValueClass();
+              } else {
+                char[] arr = p.getName().toCharArray();
```

Comment:
On the JS side we have foam.util.String.capitalize(). I think it would make the code easier to understand if we add a similar function to foam.util.StringUtil.java on the java side.